src/test/java/org/candlepin/resource/OwnerResourceTest.java

```diff
@@ -2005,24 +2006,21 @@ public void testReturnProductSysPurposeValuesForOwner() throws Exception {
         when(this.mockOwnerCurator.getByKey(eq(owner.getKey())))
             .thenReturn(owner);
 
-        CandlepinQuery mockQuery = mock(CandlepinQuery.class);
-        when(this.mockOwnerProductCurator.getProductsByOwner(eq(owner)))
-            .thenReturn(mockQuery);
+        Map<String, Set<String>> mockMap = new HashMap<>();
+        when(this.mockOwnerProductCurator.getSyspurposeAttributesByOwner(eq(owner)))
+            .thenReturn(mockMap);
 
-        Product p1 = TestUtil.createProduct();
-        Product p2 = TestUtil.createProduct();
-        Product p3 = TestUtil.createProduct();
-
-        p1.setAttribute(SystemPurposeAttributeType.ADDONS.toString(), "hello, world");
-        p2.setAttribute(SystemPurposeAttributeType.ADDONS.toString(), "hello, earth");
-        p3.setAttribute(SystemPurposeAttributeType.ADDONS.toString(), "earth, world");
-
-        p1.setAttribute(SystemPurposeAttributeType.USAGE.toString(), "production");
-        p2.setAttribute(SystemPurposeAttributeType.USAGE.toString(), "production");
-        p3.setAttribute(SystemPurposeAttributeType.USAGE.toString(), "development");
+        Set<String> mockAddons = new LinkedHashSet<>();
+        mockAddons.add("hello");
+        mockAddons.add("world");
+        mockAddons.add("earth");
+        mockMap.put(SystemPurposeAttributeType.ADDONS.toString(), mockAddons);
 
-        List<Product> dummyProducts = new ArrayList<>(Arrays.asList(p1, p2, p3));
-        when(mockQuery.list()).thenReturn(dummyProducts);
+        Set<String> mockUsage = new LinkedHashSet<>();
```

Comment:
Don't use a LinkedHashSet anymore here, or the equality check will fail if the LHS is the object doing the comparison and the order happens to be "wrong."